tests/cpp/test_persistent_buffer.cpp

```diff
@@ -1363,4 +1363,52 @@ TEST_F(PersistentBufferTest, GetResolutionIssue1123) {
       std::vector<TensorView*>{tv7});
 }
 
+TEST_F(PersistentBufferTest, InnerPersistentNotEnoughSharedMemory) {
```

Comment:
Please make sure the intended behavior indeed happens. This test could just run successfully even when, for example, it's segmented.

Reply:
This one segments on A100 since there is not enough shared memory. Do you mean https://github.com/NVIDIA/Fuser/issues/3777?

Reply:
Then, it should validate the fusion is indeed segmented. Also, please include some comment describing what this test is meant to do.

Reply:
Added comments about this test and validation of segmentation and shared memory persistence.